### PR TITLE
[FrameworkBundle] Add completion feature on translation:update command

### PR DESCRIPTION
**Pull request by stephenkhoo**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Part of #43594
| License       | MIT
| Doc PR        | -

Adding completion for translation:update.
- [x] locale
- [X] bundle 
- [X] --format
- [X] --domain
- [X] --sort

Test for 
- [x] locale
- [x] bundle 
- [X] --format
- [X] --domain
- [X] --sort

Locale completion still under discussion in https://github.com/symfony/symfony/pull/43644#pullrequestreview-786510272

Locale and bundle test still not complete

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandCompletionTest.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        $suggestions = $tester->complete($input);
+
+        $this->assertSame($expectedSuggestions, $suggestions);
+    }
+
+    public function provideCompletionSuggestions()
+    {
+        yield 'bundle' => [['en', ''], ['BaseBundle']];
+        yield 'domain' => [['en', '--domain=m'], ['messages']];
+        yield 'format' => [['en', '--format='], ['xlf12', 'xlf20']];
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationUpdateCommandCompletionTest.php]: According to the current output in case of wrong format:
> Supported formats are: php, xlf, po, mo, yml, yaml, ts, csv, ini, json, res, xlf12 and xlf20.

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 2]
```diff
@@ -321,6 +305,50 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+        if ($input->mustSuggestArgumentValuesFor('bundle')) {
+            $suggestions->suggestValues(array_map(function (BundleInterface $bundle) {
+                return $bundle->getName();
+            }, $kernel->getBundles()));
+
+            return;
+        }
+
+        if ($input->mustSuggestOptionValuesFor('format')) {
+            $suggestions->suggestValues(array_keys(self::FORMATS));
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]:
```suggestion
            $suggestions->suggestValues(array_merge(
                $this->writer->getFormats(),
                array_keys(self::FORMATS)
            ));
```

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 3]
```diff
@@ -321,6 +305,50 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+
+            return;
+        }
+
+        if ($input->mustSuggestOptionValuesFor('domain')) {
+            $locale = $input->getArgument('locale');
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: If the `locale` argument is not set, is it possible to fallback to the default locale? It is required to call the methods after.

**Review by GromNaN — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php (hunk 3 above)
Comment (stephenkhoo) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: > It is required to call the methods after.

Do you mean after the if statement?

I'm thinking because it is only used within, that's why put the 
`$locale = $input->getArgument('locale');`
after the if statement.

---
It is possible to fallback to the default locale, but the locale input is required, and I think default locale has another purpose?

Comment (GromNaN) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: If someone type `bin/console translation:update --domain ` the completion enter in this `if` but the value for `locale` is not set.
It may be a edge case that we chose to ignore and suggest nothing.

Comment (stephenkhoo) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: Yes, I think this is more reasonable, because the locale is required, so it makes more sense to only suggest if locale is typed, and we ignore and suggest nothing if locale is not typed in.

So we agree to proceed with
`if ($input->mustSuggestOptionValuesFor('domain') && $locale = $input->getArgument('locale'))` right?

**Comment by stephenkhoo** [comment]
Added locale completion with `Locales::getLocales()`

will update again if there's other conclusion from https://github.com/symfony/symfony/pull/43644#pullrequestreview-786510272

Otherwise, it seems this PR has achieved its goal?

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 4]
```diff
@@ -321,6 +308,53 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         return 0;
     }
 
+    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
+    {
+        if ($input->mustSuggestArgumentValuesFor('locale')) {
+            $suggestions->suggestValues(Locales::getLocales());
```
Comment (welcoMattic) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: Same comment as here: https://github.com/symfony/symfony/pull/43644/files#r734297130

TL;DR : it's a huge list, we must reduce it to the currently enabled locales of the application.

Comment (stephenkhoo) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: Yeap, just saw the conclusion from there, fixed it with DI for enabled locales already

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php [hunk 5]
```diff
@@ -321,6 +309,53 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+            $suggestions->suggestValues($this->enabledLocales);
+
+            return;
+        }
+
+        /** @var KernelInterface $kernel */
+        $kernel = $this->getApplication()->getKernel();
+        if ($input->mustSuggestArgumentValuesFor('bundle')) {
+            $suggestions->suggestValues(array_map(function (BundleInterface $bundle) {
+                return $bundle->getName();
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: We should probably also support bundle aliases here (see https://github.com/symfony/symfony/pull/43682/files#diff-ab7b4bb02fe1986f66a5bde7a77b924f1d86cc26d209c4399e80e8d7ee9554aaR174).

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php (hunk 5 above)
Comment (stephenkhoo) [reply src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php]: Noted on this, thank you

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @stephenkhoo.
